### PR TITLE
(phi0474.phi031) Epidoc and Capitains

### PR DESCRIPTION
**Pull request by PonteIneptique**
Hi @lcerrato ,
Outside of teiHeader and the `__cts__.xml` and the indentation, is it the kind of thing you expect ? if so, I have a nice suite of regex to apply, I could finish the Cicero in section only.

### CODE REVIEW

**Comment by capitains-hook** [comment]
## Global

| Changed            |   Status |
|:-------------------|---------:|
| `coverage`         |     0.12 |
| `metadata_passing` |     1    |
| `metadata_total`   |     1    |
| `nodes_count`      |   105    |
| `texts_passing`    |     1    |

## Words

| Changed   |   Status |
|:----------|---------:|
| `lat`     |   +10698 |

## Units

| Changed                                                 | Status   |
|:--------------------------------------------------------|:---------|
| `./data/phi0474/phi031/__cts__.xml`                     | New      |
| `./data/phi0474/phi031/__cts__.xml`                     | Passing  |
| `./data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml` | Passing  |

*[Hook UI build recap](https://ci.perseids.org/repo/PerseusDL/canonical-latinLit/5218)*

**Review by lcerrato — COMMENTED** [review]
For regularization purposes—
The title is the canonical work title.
The edition label is what appears in the print edition.
We prefer to review these separately for consistency as there are lots of metadata issues to clean up.

**Comment by lcerrato** [comment]
@PonteIneptique 
I can't review this right away — I'm still dealing with a backlog of regular work from over the break (and I sort of dropped everything yesterday because I thought you needed one text only....)

I have to get the file offline to review (I can only do inline comments on the last version of the text it seems and my comments extend to other changes.)

There are a few procedural things we like to do, such as bumping the URN (which is a CITE table step as well) and a comprehensive header review. The old Perseus data has a great deal of bad info so as part of the conversion process we need to check stuff for consistency.

This is doc from a pilot project on converting the Plutarch. It is not fully updated, but has some of the info we used.
https://docs.google.com/document/d/18fgMUufWKOcy_QTDpsJxAirZkH1c_biiS5DJs6uDVzY/edit?usp=sharing

My biggest preference from doing that set of work would be to 1) create all metadata first, review and ingest that 2) one pull request per edition file

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [inline data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:6]: The title should be Latin, yes? (whatever appears on the print page).

**Comment by lcerrato** [comment]
Ok, the header needs to look like the one I did yesterday. There are inconsistencies in the title, editor, monograph section, etc. And unnecessary stuff there, too. 
The monogr is going to match yesterday's version (if it is the same volume) so that should be easy.

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [reply data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:6]: See the text I did yesterday for these points. 

Subtitle <title>Pro Milone, Pro Marcello, ...</title> and <title type="sub">Machine readable text</title> are not needed. Strike these.

n="Cic." not needed. strike.

<editor role="editor" n="OCT">Albert Clark</editor> s/b <editor>Albert Curtis Clark</editor> (this is how his name appears on the page. and editor role editor is unnecessary)

funder n="org:NEH">National Endowment For The Humanities (NEH)</funder> s/b funder n="org:NEH">National Endowment for the Humanities</funder> (we don't capitalize it this way — again, trying to be consistent with existing collection)

The whole <monogr> section should read like the one I did yesterday. (So there are issues with the languages, with the title, missing scan, etc.)

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [inline data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:59]: All of the old refState stuff seems unnecessary. We either strike or comment out.

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [inline data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:89]: I struck the time stamp data — too much — and the n attribute?? (there are duplicates in the n attribute anyhow). Just sort newest to oldest and it will be fine.
Make sure to add your changes here.

Empty logs are not needed such as `<change n="1.4" when="1997-08-18T14:54:26" who="textgod">???</change>`

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [inline data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:123]: wrong lang id la = lat

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [inline data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:88]: wrong langUsage
`<language ident="la">Latin </language>` la s/b lat						
`<language ident="greek">Greek </language>` greek s/b grc

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [reply data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:123]: see above — this is a common issue

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [inline data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:127]: I would put the first milestone after the first div

**Comment by lcerrato** [comment]
In general, if an issue is not created up front, we are going to be unable to track unfinished business (URN bumping) or something like quotation marks, which I see here. 
If you don't want to deal with quote/q in this pass (which is a strong preference) then we need to have issues created first in order to keep these open for later fixing.

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [inline data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:708]: quotes need to be captured. `<q>` for simple speech, emphasis, inline stuff and `<quote>` for quotations (mainly block quotes)

quotes that span its containers employ rend="merge" to reflect the continuation

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [inline data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:721]: Beta code?

All of the Greek in these old files is going to be beta code and needs to be converted to Unicode. But, there are problems with those conversion scripts so be aware of those.

**Comment by PonteIneptique** [comment]
@lcerrato I needed really the one from yesterday, and I thank you for that, really.

I just wanted to see if I got things right and help, like I used to (I definitely converted my fair share, but I am always hoping to see this canonical corpus finally in Capitains one day !)

Thanks for the pointer. I am not sure I'll be able to be as thorough as you are on teiHeader, and I'll try to keep in mind the `__cts__.xml` thing but it is indeed something that needs to be curated. I am just trying to do the heavy semi-automatic lifting and help that way :)

File: data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml
Comment (lcerrato) [inline data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml:741]: Things to be aware of:
notes are sometimes not corrected tagged (wrong language). I think these are ok as they appear to be app crit in Latin but most are not and many need language attributes

I think the rend="italic" is better than "italics". 

(Again, most `<hi>` tags are meaningless. We generally like to address these by giving them semantic meaning but understandable if not possible or preferable here.)

**Review by lcerrato — DISMISSED** [review]

**Comment by lcerrato** [comment]
@PonteIneptique 
I added some comments, mostly just things I look for as I read through a text. There may be things I missed.

The main difficulty is that I cannot know where the problems are until I look at a text set or corpus. Each one has unique problems and inconsistencies. 

One way we know which is the default CTS container is the Oxford Latin Dictionary front matter. I have a scanned version here, (fortunately, since I can't get to the book). So that is how we choose between chapter or section in this instance. But every work is unique.

Unfortunately, I do not find batch conversion helpful with most Perseus texts. There are so many irregularities that these really do benefit from at least a quick glance at the content. So much of this Latin was done via bad OCR that there are many errors.

Also, BIG PROBLEM — there were about five or six wrong section breaks I found yesterday. It is really important to at least search for hard breaks at the ends of the divs — if there is no hard break there, it could still be right (only a comma for instance) but in most cases, it is wrong and the script missed the break entirely.  This is was very important in most of the previous Latin prose (such as earlier Cicero and Caesar).

Checking that there is a hard break at the end of the section is really the big thing we have to do with this Latin. Most of these breaks were not entered by hand.

**Comment by lcerrato** [comment]
@mkonieczny9805
Would you be interested in assisting with this conversion batch? (I actually don't know how many Thibault would like to complete)
These have issues similar to the Plutarch overall — I'm sure there is other stuff lurking in here. (Inconsistent conversion of v to u, bad OCR, etc.) I started the Cicero years ago with a student, so even those early works are not as up to date in terms of header and cts metadata consistency.

I did one text yesterday 474.27 which could serve as an example — sort of in a hurry — but I'd appreciate another set of eyes on this. 

I think one thing that would be good would be to create the cts work metadata upfront (as opposed to how we did it with Plutarch). Even though there are lots of works, there aren't that many volumes.

Let me know if you are interested.

**Comment by mkonieczny9805** [comment]
Sure, I could do some work on this. I assume we're only working on the texts not yet on Scaife? So, 474.20-23 and 26-34? Looks like 13 works total.

**Comment by PonteIneptique** [comment]
I could do a text a day, on European work hours

**Comment by lcerrato** [comment]
@mkonieczny9805 
Yes. I suspect there are inconsistencies with the earlier works, particularly in the metadata — but an issue placeholder will suffice there.
If you don't want to open issues for each work, then don't bother. I just need to remember to bump the Ids.
I'll create one issue for the new cts work level metadata files, though and get to work on those first. 
As that may create a conflict, let's hold this one for now.

Note: We use whatever container the OLD front matter indicates. 
So for in Pisonem we used the sections (99 of them) rather than the chapters, as shown here:
image

I have the OLD front matter scanned in case you need it. This is a sticking point in the Cicero especially.

**Comment by lcerrato** [comment]
See https://github.com/PerseusDL/canonical-latinLit/issues/380

Note we're just focusing on Latin for now.

**Comment by lcerrato** [comment]
@mkonieczny9805 
Hi yes, I want to create the CTS files and push them. 

Then you will be reviewing the files Thibault converts. 
You can either just review his work or divide them up, or whatever other plan you think works. If you'd prefer to do it in stages, we can do that, too. (Push his changes and then review them in another workflow/batch).

Otherwise, these are going to sit here for longer than I'd like —  I know I cannot review the whole corpus right away. The data is not very good, so there were lots of things I had to double check when I worked on Piso yesterday.

**Comment by lcerrato** [comment]
@mkonieczny9805 
I sent you an invite as a new collaborator so you should have the ability to push changes.

**Comment by mkonieczny9805** [comment]
@lcerrato @PonteIneptique 
Great. Since the data is bad, doing the work in stages makes the most sense to me. Thibault will do a first round of edits, then I'll push his changes and review. Perhaps I can start by quickly double-checking 474.27.

**Comment by lcerrato** [comment]
@mkonieczny9805 @PonteIneptique 
I checked in a batch of cts work files for this. Some of these sources are odd so we won't be updating all of the Perseus Latin. If there is anything we haven't already added to OGL (and there might be gaps) we can think about new sources.

A pull from the repo needs to be done to avoid conflicts (or my changes need to be added to this issue).

I would suggest maybe fresh PRs.

**Comment by PonteIneptique** [comment]
@lcerrato I merged your update :)

As for the implication of your message, do you know which we should not convert to Capitains ?

**Comment by capitains-hook** [comment]
## Global

| Changed         |   Status |
|:----------------|---------:|
| `coverage`      |     0.09 |
| `nodes_count`   |   105    |
| `texts_passing` |     1    |

## Words

| Changed   |   Status |
|:----------|---------:|
| `lat`     |   +10698 |

## Units

| Changed                                                 | Status   |
|:--------------------------------------------------------|:---------|
| `./data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml` | Passing  |

*[Hook UI build recap](https://ci.perseids.org/repo/PerseusDL/canonical-latinLit/5224)*

**Comment by lcerrato** [comment]
@PonteIneptique @mkonieczny9805 
There won't be any files here for the ones that cannot be converted. You will see `.json` files but no xml files in the directory. Otherwise, if there is a question, we can deal with it as it goes. If the publication year is after 1925 then it needs a second look.
Many of the Perseus Latin texts are poorly sourced, but the Cicero should be ok.

**Comment by PonteIneptique** [comment]
Great, so if I get things right:

- I convert to have the general structure and PR
- @mkonieczny9805 checks the content (including quote and missed information), upgrade and merge ?

**Comment by mkonieczny9805** [comment]
@PonteIneptique @lcerrato 
Yes I think that sounds like a good workflow, although I'm not sure I'm authorized to merge.

Btw I had a look at 474.27 and did not notice any obvious issues, except that in both the text and CTS files the title should be "M. Tulli Ciceronis Orationes" without a period after "Ciceronis." Should I open a separate PR for this change or can it be incorporated into this one? In the latter case we would once again need to merge this PR with Master, or open a new PR.

I had a few questions about the XML though:
1. I recall that in the Plutarch conversion work `milestone` elements were supposed to be contained within `p` elements. I noticed `milestone` elements outside of `p` elements in this file, and wonder if that should be changed.
2. This is the first time I've seen sentence-initial words enclosed within `reg` tags. Is this a holdover from the old files or are they a requirement moving forward?
3. Noticed a `textpart` container separating "fragments" from "speech" that is not included in the refsDecl and wonder whether that is standard practice.

**Comment by lcerrato** [comment]
@mkonieczny9805 
You should be able to merge now.
I used M. Tulli Ciceronis Orationes as the book plate title (based on WorldCat) as Cicero is already added there as the short author name. If it's wrong there, then other CTS files may be the same (since they are all cut and paste). I wouldn't spend time fixing this as we can do that within the GUI in seconds.

1. milestones are better inside the divs when possible, yes.

2. `reg` is there because the editor changed uppercase to lowercase. (That is in the source file and there are capitalized words that were missed, so I suspect this was a scripted thing.)

3. The refsDecl is just looking for sections, so it will ignore that distinction between fragment and speech (unless there is an error there). As long as the sections are uniquely named. This is something that we do elsewhere, such as drama, where there are divs such as episode, chorus, strophe, but the container is still lines.

**Comment by mkonieczny9805** [comment]
@lcerrato I am unable to locate the 1918 edition (2nd ed) on Internet Archive. According to the header this was the source text. Any advice on what to put down for the `ref target` attribute?

**Comment by mkonieczny9805** [comment]
@lcerrato 
Another question I have is, do we want to keep the "notes" derived from the app crit? It seems that not all Cicero texts have them, but Pro Milone does.

**Comment by lcerrato** [comment]
@mkonieczny9805 
I think I found most all of the editions except the copyrighted ones — some were available to borrow only and many were listed with variant titles. I can double check that.
The years may be wrong in some of these but I would have corrected them in the cts files.
Let me double check.

**Comment by lcerrato** [comment]
@mkonieczny9805 
If something has the notes, I never remove them. It is probably not efficient to add notes where they were not included, however. 
Fixing errors in the notes can be time consuming, however, so a notation in the editorialDesc about any issues with note quality is helpful if you think they are a mess.

**Comment by lcerrato** [comment]
@mkonieczny9805 
https://archive.org/details/orationes0002cice/page/n5/mode/2up

**Comment by mkonieczny9805** [comment]
@lcerrato 
HathiTrust seems better since I can link directly to the page. However, their permanent url causes a problem in the XML file because it contains the string "view=", which Oxygen interprets as an attribute. I think we've run into this before and I don't remember the solution.

**Comment by lcerrato** [comment]
@mkonieczny9805 
The left sidebar has a permalink, I think. ("Link to this page")

**Comment by mkonieczny9805** [comment]
@lcerrato 
That's what I've tried using, and that's the link that has the problematic string. I don't suppose we could use a TinyURL or something?

**Comment by PonteIneptique** [comment]
@mkonieczny9805 Can you show me what Oxygen is refusing ? normally, `=` is a valid character :)

**Comment by lcerrato** [comment]
@mkonieczny9805 
I'm a couple of texts now from HT and that doesn't seem to be happening. I know I saw it in the past, though.

This is an active text here:
https://github.com/OpenGreekAndLatin/First1KGreek/blob/master/data/tlg0018/tlg008/tlg0018.tlg008.opp-grc1.xml

**Comment by mkonieczny9805** [comment]
@lcerrato @PonteIneptique 
Ok, replacing & with ; in the automatically generated URL solves the problem and still links to the proper page:

https://babel.hathitrust.org/cgi/pt?id=uc1.31822018932749;view=2up;seq=38

becomes

https://babel.hathitrust.org/cgi/pt?id=uc1.31822018932749;view=2up;seq=38

**Comment by PonteIneptique** [comment]
Ohh ! The `&` is indeed a special char, you can replace it with the correct `&amp;` string :)

**Comment by mkonieczny9805** [comment]
Let's file it away for future reference!

**Comment by mkonieczny9805** [comment]
@lcerrato @PonteIneptique 
I checked all the section breaks and found maybe 7-8 incorrect ones. One notable error is that when an abbreviated praenomen appears in the line, the period is treated as the end of the sentence, so if the section break falls in that line the break has sometimes been inserted between the praenomen and the rest of the name.

There were a few weird "soft" stops but that's how they're printed in Clark, so I kept them as is.

**Comment by mkonieczny9805** [comment]
Also noticed that this text has a lot of quotations which have to be replaced with `<q>` tags. But some of them are long and straddle section breaks. @lcerrato can you remind me how to handle those?

**Comment by lcerrato** [comment]
@mkonieczny9805 
Not surprising on the breaks being incorrect. The script doesn't know what a praenomen is .... :)

Some soft breaks may not be all that soft — I noticed a couple of instances where the Loeb had included a semi-colon where Oxford had a comma, or something like that. Commas are ok as a fall back. There are rare instances where the break is just wrong. Sometimes other marginalia bump the digits to another line. 
If all else fails and you think it's a terrible place to have a `div` you can move the break and just add a `@resp="perseus"` attribute to the new section. Keep the old one as a milestone or make a note.

`rend="merge"` should appear at the start of the next container when a quote or quotation straddles the containers.

So first instance `<q>`, and then, following the `<div>` break,  `<q rend="merge">`

**Comment by mkonieczny9805** [comment]
@lcerrato @PonteIneptique 
I've just about finished going through 474.31. The notes (i.e., app crit) actually look very good except that the Greek needs to be converted to Unicode, and I don't know how to do that automatically. In addition, the superscripts appear just as regular numerals.
A question about the milestones: when chapter breaks coincide with section breaks, the milestones appear at the end of the preceding section rather than the beginning of the new one as in 474.27. Should this be changed?

**Comment by lcerrato** [comment]
@mkonieczny9805 
1. We have some jar files on hand to convert the Greek but there are sometimes errors it can introduce. I use tei.transformer.lang_grc.jar as described here:
https://github.com/PerseusDL/tei-conversion-tools/wiki/Greek-Betacode-to-Unicode-Transformations

2. Yes, ideally, those would be moved for readability. Sometimes there are regex that can do it if the pattern is consistent. I would not spend much time on it. 

If you would like to leave any of this for someone else, we can always create an issue at the end of the batch that addresses the problems.

**Comment by mkonieczny9805** [comment]
@lcerrato I'll have a stab at it

**Comment by mkonieczny9805** [comment]
@PonteIneptique @lcerrato 
The .jar transform seemed to work perfectly except that it does not distinguish between Σ and ς, so I fixed that with a find/replace. Pushed my revisions and awaiting completion of the hook test (it should pass since I ran it on the test repo that Thomas created). Lisa it might be a good idea for you to have a quick look that there aren't any glaring issues remaining.

**Comment by capitains-hook** [comment]
## Global

| Changed         |   Status |
|:----------------|---------:|
| `coverage`      |     0.09 |
| `nodes_count`   |   105    |
| `texts_passing` |     1    |

## Words

| Changed   |   Status |
|:----------|---------:|
| `lat`     |   +10684 |

## Units

| Changed                                                 | Status   |
|:--------------------------------------------------------|:---------|
| `./data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml` | Deleted  |
| `./data/phi0474/phi031/phi0474.phi031.perseus-lat2.xml` | New      |
| `./data/phi0474/phi031/phi0474.phi031.perseus-lat2.xml` | Passing  |

*[Hook UI build recap](https://ci.perseids.org/repo/PerseusDL/canonical-latinLit/5229)*

**Review by mkonieczny9805 — COMMENTED** [review]
Reviewed by MK as a test of the review feature

**Comment by lcerrato** [comment]
@mkonieczny9805 

Great! I took a quick look and I can't spot anything at the moment. I think you added a .DS_Store file to the PR? That's the only thing I spotted, but 

With the transform you have to watch out with the transform for closed parentheses in lieu of apostrophe. beta code `)` versus `'` and it does try insert medial sigma at the end of words.

This is such a help.

**Review by lcerrato — APPROVED** [review]

**Comment by mkonieczny9805** [comment]
Awesome. That .DS_Store file was automatically generated and I'm not sure what it does. Do we not need it? Can it be removed from the PR?

I looked out for those issues with the transform but most of the Greek characters here were just sigla so I think we're ok.

**Comment by mkonieczny9805** [comment]
Also there are three options for merging. Which one do you usually use?

**Comment by lcerrato** [comment]
@mkonieczny9805 
.DS_store is computer generated binary file and we don't want it in the data.

I just use the green button (nothing on the drop down).

**Comment by capitains-hook** [comment]
## Global

| Changed         |   Status |
|:----------------|---------:|
| `coverage`      |      0.1 |
| `nodes_count`   |    105   |
| `texts_passing` |      1   |

## Words

| Changed   |   Status |
|:----------|---------:|
| `lat`     |   +10684 |

## Units

| Changed                                                 | Status   |
|:--------------------------------------------------------|:---------|
| `./data/phi0474/phi031/phi0474.phi031.perseus-lat1.xml` | Deleted  |
| `./data/phi0474/phi031/phi0474.phi031.perseus-lat2.xml` | New      |
| `./data/phi0474/phi031/phi0474.phi031.perseus-lat2.xml` | Passing  |

*[Hook UI build recap](https://ci.perseids.org/repo/PerseusDL/canonical-latinLit/5232)*

**Comment by mkonieczny9805** [comment]
@PonteIneptique @lcerrato 
So what's next?

**Pull request merged**
